src/libraries/System.Text.Json/gen/Reflection/RoslynExtensions.cs

```diff
@@ -33,6 +33,13 @@ public static IEnumerable<INamedTypeSymbol> BaseTypes(this INamedTypeSymbol type
             }
         }
 
+        /// <summary>
+        /// Creates a copy of the Location instance that does not capture a reference to Compilation.
+        /// </summary>
+        [return: NotNullIfNotNull(nameof(location))]
+        public static Location? GetTrimmedLocation(this Location? location)
+            => location is null ? null : Location.Create(location.SourceTree?.FilePath ?? "", location.SourceSpan, location.GetLineSpan().Span);
```

Comment:
We've now duplicated this in several of our generators. @chsienki, should Roslyn include helpers for this (and maybe more generally on a Diagnostic so that someone writing a source generator can easily avoid referencing the compilation from a Diagnostic they hold onto)?

Reply:
Yeah, that seems like a good candidate for a new API